tests/framework/common_apps.go

```diff
@@ -170,6 +171,9 @@ func (td *OsmTestData) SimplePodApp(def SimplePodAppDef) (corev1.ServiceAccount,
 		Spec: corev1.PodSpec{
 			TerminationGracePeriodSeconds: new(int64), // 0
 			ServiceAccountName:            def.Name,
+			NodeSelector: map[string]string{
```

Comment:
For this function and the one for SimpleDeploymentAppDef, I think you could just a local variable to be the DefaultOS if the string is empty, and then take the value if it is not. That way you don't have to modify all of the other tests to pass this in.

Reply:
No strong opinions though

Reply:
Images can't be cross platform across Linux and Windows though, unless something has changed. Linux nodes will always have different images than Windows nodes

Reply:
> Images can't be cross platform across Linux and Windows though,

I meant that in the sense that there is a corresponding windows image. Not that the same image will be used in both platforms.

Reply:
I think modifying the tests for hybrid traffic will require a bit more planning/discussion, but I definitely see your point.

Since we're not at a stage where we are testing hybrid traffic yet (as of this PR), can we scope this PR to not modify the individual tests and keep the login in common_apps? That way the changes are more localized and changes can be made easily as you iterate in the future.

Reply:
fixed with the latest update. I hope that you meant to add a return `err` value that tests should verify